aws/vpc/main.tf

```diff
@@ -89,7 +89,28 @@ resource "aws_nat_gateway" "nat_gateway" {
 }
 
 /*
- * Routes for private subnets to use NAT gateway
+ * Set to true create transit gateway attachments in the private subnets and
+ * route traffic to the TGW instead of NAT GW.
+ * Should be used when create_nat_gateway=false.
+ */
+#Create Transit Gateway VPC attachment for orlando-dr to attach to hub.
+resource "aws_ec2_transit_gateway_vpc_attachment" "transit_gateway" {
+  count = var.create_transit_gateway_attachment ? 1 : 0
```

Comment:
This is just a suggestion, but it might be clearer to name this for the bigger-picture goal, not specifically this implementation detail. What would you think of renaming the `create_transit_gateway_attachment` variable to something like `use_transit_gateway`?

Then, if a workspace sets `use_transit_gateway` to `true`, the terraform code will do the things necessary to make that happen, which currently include creating an `aws_ec2_transit_gateway_vpc_attachment` and an `aws_route` to the transit gateway. Those implementation details might change later, but the variable name wouldn't need to because the intent remains the same.

Reply:
I changed it to `use_transit_gateway`.  Not sure if it should go further and disable the NAT gateway when transit gateway is used or just assume the administrator would also include `create_nat_gateway=false`  I'm not sure if there's a use case where a  NAT gateway would still be needed.  But we definitely can't have two opposing route statements at the same time.